packages/fetchai/skills/erc1155_client/handlers.py

```diff
@@ -150,7 +151,7 @@ def _handle_propose(self, msg: FipaMessage, dialogue: Dialogue) -> None:
             )
             contract = cast(ERC1155Contract, self.context.contracts.erc1155)
             contract.set_address(
-                ledger_api=self.context.ledger_apis.ethereum_api,
+                ledger_api=self.context.ledger_apis.get_api(ETHEREUM),  # type: ignore
```

Comment:
I had to add `type: ignore` here and in analogous places because `contract.set_address` required `EthereumApi`. I preferred this apporach to `cast(EthereumApi, obj)` because we would have to import the class `EthereumApi`, hence increasing coupling.

Ideally, the contract should work with any ledger api, at design level.